Docs: add addon registry (#731)
oam-dev/kubevela.io: docs/platform-engineers/addon/addon-registry.md

---
title: Build your addon registry
---

An addon registry can be used for discovering and  distributing addon. Currently,KubeVela has supported two types registry: git and helm repo.

## Git as registry

A directory contains some sub-directories stored in git repository can be used as an addon registry.

The git type registry  type has  supported github, gitlab and gitee.

A typical git addon registry is like [catalog](https://github.com/kubevela/catalog/tree/master/addons). You can clone this repo to your local path and then push to your own git repository.

Then you can use this command to add your addon registry.

```yaml
vela addon registry add my-repo --type git --endpoint=<URL> --path=<ptah> --gitToken=<git token>
```

> 

Reviewer: what about the private git repo?

## Helm repository as registry

> A helm repository can be used store addon package. If you already have a helm repo, you can add it as addon registry by this command:

Reviewer: what does a "helm repostory" mean? How to build the "helm repostory"?

```yaml
vela addon registry add my-repo --type helm --endpoint=<URL>
```

> If your helm repo setup with http basic auth, you can set the username and password by:

Reviewer: can you explain or give some links about the "basic auth" of helm repo?

```yaml
vela addon registry add my-repo --type helm --endpoint=<URL> --username=<username> --password=<passwor>
```

> A helm repo type registry can store addon's multi-versions of an addon. In the future we will support chart-museum addon, and provide an  easier way to build your own versioned addon registry.

Reviewer: what does chart-museum mean here?

---
title: Build your addon registry
---

> An addon registry can be used for discovering and distributing addon. Currently,KubeVela has supported two types

Reviewer: ```suggestion
An addon registry can be used for discovering and distributing addons. Currently, KubeVela has supported two type
```

你装个 vscode markdown 插件吧，会自动检查格式。

> registry: git and helm repo.

Reviewer: ```suggestion
of registry: git server and helm repo.
```

## Git as registry

A directory contains some sub-directories stored in git repository can be used as an addon registry.

The git type registry type has supported github, gitlab and gitee.

A typical git addon registry is like [catalog](https://github.com/kubevela/catalog/tree/master/addons). You can clone
this repo to your local path and then push to your own git repository.

> If your repository type is github,you can use this command to add your addon registry.

Reviewer: ```suggestion
If your repository type is github, you can use this command to add your addon registry.
```

Please refine the overall format.

```yaml
vela addon registry add my-repo --type git --endpoint=<URL> --path=<ptah> --gitToken=<git token>
```

> If is gitee, you can use:

Reviewer: ```suggestion
If your type is gitee, you can use:
```

语法

```yaml
vela addon registry add my-repo --type gitee --endpoint=<URL> --path=<ptah> --gitToken=<git token>
```

> If is gitlab, you can use:

Reviewer: 同上

```yaml
vela addon registry add my-repo --type gitlab --gitRepoName=<repoName> --endpoint=<URL> --path=<ptah> --gitToken=<git token>
```

> 

Reviewer: 这里也要讲一下怎么上传一个 addon。git push 之类的得有。

## Helm chart repository as registry

A [helm chart repository](https://helm.sh/docs/topics/chart_repository/) can be used to store addon package.

> You can build your own helm repository follow
> this [tutorial](https://helm.sh/docs/topics/chart_repository/#hosting-chart-repositories).

Reviewer: ```suggestion
You can build your own helm repository by following
this [tutorial](https://helm.sh/docs/topics/chart_repository/#hosting-chart-repositories).
```

> If you already have one, you can add it as addon registry by this command:

Reviewer: 怎么就直接可以addon了，里面的addon怎么放进去的？

Use [helm cli](https://helm.sh/docs/intro/install/#helm) to push addon package which generated by last step to the helm repository.

```yaml
$ helm plugin install https://github.com/chartmuseum/helm-push
$ helm cm-push <addon-pacakge> <remote>
```

> More info please [refer](https://github.com/chartmuseum/helm-push).

Reviewer: ```suggestion
More info please refer to [chartmuseum docs](https://github.com/chartmuseum/helm-push).
```
